pkg/controller/containerjfr/containerjfr_controller_test.go

```diff
@@ -201,7 +193,71 @@ var _ = Describe("ContainerjfrController", func() {
 				Expect(result).To(Equal(reconcile.Result{}))
 			})
 		})
+		Context("Switching from a minimal to a non-minimal deployment", func() {
+			BeforeEach(func() {
+				objs = []runtime.Object{
+					test.NewMinimalContainerJFR(),
+				}
+			})
+			JustBeforeEach(func() {
+				reconcileFully(client, *controller, true)
```

Comment:
Sorry I didn't notice this sooner, could you change your helper functions to take a pointer to `containerjfr.ReconcileContainerJFR`? This is more efficient, since otherwise a copy of the whole struct is created for each call. In general, you will want to pass structs by pointer, unless you're concerned about side-effects.